src/lib/viewers/doc/DocBaseViewer.js

```diff
@@ -334,6 +339,23 @@ class DocBaseViewer extends BaseViewer {
             .catch(this.handleAssetError);
     }
 
+    /**
+     * Checks for xrefs, depending on file extension
+     * @return {void}
+     */
+    checkForXrefs() {
+        const { extension, id } = this.options.file;
+
+        if (extension === 'dwg') {
```

Comment:
Should this live in its own viewer that extends `DocBaseViewer`? I think we generally try to allow the loader to do extension-specific stuff.